File: src/minymph_project/Shop.java
package minymph_project;

/**
 * Represents a shop in the game where players can purchase items or upgrades
 * using in-game currency.
 */
public class Shop {

    private Double money;

    /**
     * Constructs a new Shop instance with default settings.
     */
    public Shop() {
    }

    /**
     * Returns the current amount of money available in the shop.
     *
     * @return the available money as a Double
     */
    public Double getMoney() {
        return money;
    }

    /**
     * Sets the amount of money available in the shop.
     *
     * @param money the amount to set as available money
     */
    public void setMoney(Double money) {
        this.money = money;
    }
}

File: core/Shop.py
from typing import Optional

class Shop:
    """
    Represents a shop in the game where players can purchase items or upgrades
    using in-game currency.
    """

    def __init__(self) -> None:
        """
        Constructs a new Shop instance with default settings.
        """
        self._money: float = 0.0

    def get_money(self) -> float:
        """
        Returns the current amount of money available in the shop.

        Returns:
            The available money as a float.
        """
        return self._money

    def set_money(self, money: float) -> None:
        """
        Sets the amount of money available in the shop.

        Args:
            money: The amount to set as available money.
        """
        self._money = money